src/vade_sidetree.rs

```diff
@@ -992,6 +992,299 @@ mod tests {
+            patches: Some(vec![patch]),
+            next_recovery_key: None,
+            recovery_key: None,
+        };
+
+        let result = did_handler
+            .did_update(
+                &create_response.did.did_document.id,
+                &"{\"type\":\"sidetree\", \"waitForCompletion\":true}",
+                &serde_json::to_string(&update_payload)?,
+            )
+            .await;
+
+        assert_eq!(result.is_ok(), true);
+
+        // after update, resolve and check if there is the new added service
+        let result = did_handler
+            .did_resolve(&create_response.did.did_document.id)
+            .await;
+
+        let did_resolve = match result? {
+            VadePluginResultValue::Success(Some(value)) => value.to_string(),
+            _ => return Err(Box::from("Unknown Result for did resolve".to_string())),
+        };
+
+        let resolve_result: SidetreeDidDocument = serde_json::from_str(&did_resolve)?;
+
+        let did_document_services = resolve_result
+            .did_document
+            .service
+            .ok_or("No Services defined")?;
+        assert_eq!(did_document_services.len(), 1);
+        assert_eq!(did_document_services[0].service_endpoint, service_endpoint);
+
+        let patch: Patch = Patch::RemoveServices(vade_sidetree_client::RemoveServices {
+            ids: vec!["sds".to_string()],
+        });
+
+        let mut update2_key_pair = create_response.update_key.clone();
+        update2_key_pair.nonce = Some('2'.to_string());
+
+        let update_payload = DidUpdatePayload {
+            update_type: UpdateType::Update,
+            update_key: Some(update1_key_pair),
+            next_update_key: Some((&update2_key_pair).into()),
+            next_recovery_key: None,
+            patches: Some(vec![patch]),
+            recovery_key: None,
+        };
+
+        let result = did_handler
+            .did_update(
+                &create_response.did.did_document.id,
+                &"{\"type\":\"sidetree\", \"waitForCompletion\":true}",
+                &serde_json::to_string(&update_payload)?,
+            )
+            .await;
+
+        assert_eq!(result.is_ok(), true);
+
+        // after update, resolve and check if the service is removed
+        let result = did_handler
+            .did_resolve(&create_response.did.did_document.id)
+            .await;
+
+        assert!(result.is_ok());
+
+        let did_resolve = match result? {
+            VadePluginResultValue::Success(Some(value)) => value.to_string(),
+            _ => return Err(Box::from("Unknown Result for did update".to_string())),
+        };
+
+        let resolve_result: SidetreeDidDocument = serde_json::from_str(&did_resolve)?;
+        assert_eq!(resolve_result.did_document.service.is_none(), true);
+
+        Ok(())
+    }
+
+    #[tokio::main]
+    #[test]
+    #[serial]
+    async fn can_update_did_three_times_with_nonce() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
No too bad its a test, but would it make sense to outsource the duplicates into a helper function? :D

Reply:
refactored in #13